cellarium/cas/client.py

```diff
@@ -51,6 +51,8 @@ def __init__(
         self._print("Connecting to the Cellarium Cloud backend...")
         self.cas_api_service.validate_token()
 
+        self.cas_api_service.validate_version()
```

Comment:
What does this look like when it fails?  Should we catch and add instructions in the error message to update the client?

Reply:
It prints a stack trace with the message from the error message that's produced on the server side.  So assuming your version number is 1.3.0, you'll get a stack trace with this message:
```
Client version 1.3.0 is older than the minimum version for this server 1.4.0.
Please update to the latest version using 'pip install cellarium-cas --upgrade'.
```

Reply:
awesome.  That's great

Reply:
My guess is it will be HTTPError or something similar. Maybe it would provide a slightly better user experience if an error like ClientVersionError were raised instead. I think having as few HTTPErrors as possible would be better.

Reply:
That makes sense.  Maybe I can catch the error and convert it to a ClientVersionError

Reply:
Just catching doesn't sound good to me, because catching HTTPError or even HTTPError400 doesn't necessarily mean catching improper version.

Reply:
Well, on one hand if you handle 400, then I don't see any cases when something else apart from incorrect version will be handled, like it's impossible to get a validation error in this request. On the other hand it might be confusing to handle all responses with 400 status code and propagate them all as ClientVersionError.

Reply:
That makes sense.  I think I can make that work